lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -201,7 +201,8 @@ public Uri CreateUriForPath(string path)
             {
                 throw new ArgumentNullException(nameof(path), "Specified path is null");
             }
-            return new Uri($"{GetUriPrefix()}/{path.TrimStart('/')}");
+            var uriPrefix = GetUriPrefix();
+            return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix + '/' + path.Trim('/'));
```

Comment:
Please append with the built-in Uri class instead of path separator operations.
```csharp
return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix, path);
```

Reply:
Fixed.